From c2eb62f01c3e3c32d16bf9ef231132e02806dd1d Mon Sep 17 00:00:00 2001
From: Marek Kwaczynski <marek.kwaczynski@plume.com>
Date: Fri, 20 May 2022 14:52:08 +0200
Subject: [PATCH] Revert SAE: Verify that STA negotiated H2E if it claims to support it

Revert commit 898b6d58f310bc2c415548ebc45415c93461e75b
Author: Jouni Malinen <jouni@codeaurora.org>
Date:   Fri Oct 18 12:38:11 2019 +0300

    SAE: Verify that STA negotiated H2E if it claims to support it

Signed-off-by: Marek Kwaczynski <marek.kwaczynski@plume.com>
---
 src/ap/drv_callbacks.c | 15 ---------------
 src/ap/ieee802_11.c    | 11 -----------
 2 files changed, 26 deletions(-)

diff --git a/src/ap/drv_callbacks.c b/src/ap/drv_callbacks.c
index 0467a7e..77774d6 100755
--- a/src/ap/drv_callbacks.c
+++ b/src/ap/drv_callbacks.c
@@ -16,7 +16,6 @@
 #include "common/ieee802_11_common.h"
 #include "common/wpa_ctrl.h"
 #include "common/dpp.h"
-#include "common/sae.h"
 #include "common/hw_features_common.h"
 #include "crypto/random.h"
 #include "p2p/p2p.h"
@@ -482,20 +481,6 @@ int hostapd_notif_assoc(struct hostapd_data *hapd, const u8 *addr,
 			}
 		}
 #endif /* CONFIG_IEEE80211R_AP */
-#ifdef CONFIG_SAE
-		if (hapd->conf->sae_pwe == 2 &&
-		    sta->auth_alg == WLAN_AUTH_SAE &&
-		    sta->sae && !sta->sae->h2e &&
-		    elems.rsnxe && elems.rsnxe_len >= 1 &&
-		    (elems.rsnxe[0] & BIT(WLAN_RSNX_CAPAB_SAE_H2E))) {
-			wpa_printf(MSG_INFO, "SAE: " MACSTR
-				   " indicates support for SAE H2E, but did not use it",
-				   MAC2STR(sta->addr));
-			status = WLAN_STATUS_UNSPECIFIED_FAILURE;
-			reason = WLAN_REASON_UNSPECIFIED;
-			goto fail;
-		}
-#endif /* CONFIG_SAE */
 	} else if (hapd->conf->wps_state) {
 #ifdef CONFIG_WPS
 		struct wpabuf *wps;
diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index bcd3f0e..1c880d5 100755
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -3471,17 +3471,6 @@ static int check_assoc_ies(struct hostapd_data *hapd, struct sta_info *sta,
 				   MAC2STR(sta->addr), sta->auth_alg);
 			return WLAN_STATUS_NOT_SUPPORTED_AUTH_ALG;
 		}
-
-		if (hapd->conf->sae_pwe == 2 &&
-		    sta->auth_alg == WLAN_AUTH_SAE &&
-		    sta->sae && !sta->sae->h2e &&
-		    elems.rsnxe && elems.rsnxe_len >= 1 &&
-		    (elems.rsnxe[0] & BIT(WLAN_RSNX_CAPAB_SAE_H2E))) {
-			wpa_printf(MSG_INFO, "SAE: " MACSTR
-				   " indicates support for SAE H2E, but did not use it",
-				   MAC2STR(sta->addr));
-			return WLAN_STATUS_UNSPECIFIED_FAILURE;
-		}
 #endif /* CONFIG_SAE */
 
 #ifdef CONFIG_OWE
-- 
2.17.1

